File: smoothing/medianBluring.py
# -----------------------------------------------------------------------------
#
# Example for median bluring
#
# -----------------------------------------------------------------------------

import cv2

img = cv2.imread("../../img.jpg", 0)

# Bluring apply only even numbers
blurred_7 = cv2.medianBlur(img, 7)
blurred_5 = cv2.medianBlur(img, 5)
blurred_3 = cv2.medianBlur(img, 3)

# Then show all of them
cv2.imshow("img", img)
cv2.imshow("blurred_7", blurred_7)
cv2.imshow("blurred_5", blurred_5)
cv2.imshow("blurred_3", blurred_3)

cv2.waitKey()
cv2.destroyAllWindows()

File: smoothing/medianBluring.cpp
// -----------------------------------------------------------------------------
//
// Example for median bluring
//
// -----------------------------------------------------------------------------

#include <opencv2/opencv.hpp>

int main() {
    cv::Mat img = cv::imread("../../img.jpg", 0), blurred_3, blurred_5, blurred_7;

    // Median bluring applies only even numbers
    cv::medianBlur(img, blurred_3, 3);
    cv::medianBlur(img, blurred_5, 5);
    cv::medianBlur(img, blurred_7, 7);

    cv::imshow("img", img);
    cv::imshow("blurred_7", blurred_7);
    cv::imshow("blurred_5", blurred_5);
    cv::imshow("blurred_3", blurred_3);

    cv::waitKey();
    cv::destroyAllWindows();
}
